vllm/v1/engine/output_processor.py

```diff
@@ -39,13 +40,16 @@ class RequestOutputCollector:
     producer gets ahead of the consumer.
     """
 
-    def __init__(self, output_kind: RequestOutputKind):
+    def __init__(self, output_kind: RequestOutputKind, request_id: str):
         self.aggregate = output_kind == RequestOutputKind.DELTA
+        self.request_id = request_id
         self.output: RequestOutput | PoolingRequestOutput | Exception | None = None
         self.ready = asyncio.Event()
 
     def put(self, output: RequestOutput | PoolingRequestOutput | Exception) -> None:
         """Non-blocking put operation."""
+        if isinstance(output, (RequestOutput, PoolingRequestOutput)):
+            assert output.internal_req_id == self.request_id
```

Comment:
could put this in the corresponding elif blocks below to save an additional if (assuming we keep internal_req_id in the outputs, per other comment)

Reply:
The next line makes it a bit tricky:

```
        if self.output is None or isinstance(output, Exception):
            self.output = output
```

The assertion wouldn't apply to the `Exception` case. That's why I did it here

Reply:
right, I meant the same assert could be included in both of the `elif` blocks below that one